app/src/main/java/com/bogdan/codeforceswatcher/features/problems/redux/requests/ProblemsRequests.kt

```diff
@@ -60,8 +86,37 @@ class ProblemsRequests {
             return true
         }
 
+        private suspend fun updateDatabaseAndDispatch(newProblems: List<Problem>) {
+            val differenceProblems = mutableListOf<Problem>()
```

Comment:
Aren't we getting all problems every time? Why then doing diff here? Why not just clear database and put all records back. Measure how much time you need for this.